### ISSUE TITLE
Feature request - Change all light colours at Sunrise / Sunset

### ISSUE THREAD

**Issue by Sandown** [Hyrules/WinHue3#72]
Hi,

A feature request here. I don't think its possible at the minute although bits of it are with rules etc.

Looking to do exactly what F.lux does on my desktop (yes I realize there is a flux option for hue but it doesn't work and doesn't offer any options either) which is to change the colour / hue of all the lights at a predefined time before or after sunset / sunrise. This needs to be a phased transition that happens over 60 mins (or option to change this) and does not turn any lights on or off, simply changes the colour for the ones already powered on, and sets ready for when the other offline bulbs do come back online. I'm aware the Philips bulbs do not have any memory and default to white when powered back on so we need a workaround for this too as part of this. One way might be to push the current state of the colour transition (keep in mind it should happen over 60 minutes) to all bulbs every 2-3 seconds perhaps? Open to suggestions on that one.

Open to discussion on this :)

**Comment by Hyrules** [comment]
I know of f.lux. I have seen it in action tried it a couple of times. I understand what you want. It's doable for lights already on but not for the lights that are off. You cannot set the state of a light that is off / or unreachable. Bridge refuses it. I could develop a tool at some point. Many people have been complainig to philips about the light not retaining the state when power is turned off. It's not a killer problem for me just bugging when there is a power failure in the middle of the night. What color would the light go to ? a settable color ? or just change some hue ?

**Comment by Sandown** [comment]
Thanks for the fast reply. I really do appreciate your time on this. The light would go to a settable colour presumably (specific hue). I'd suggest the best bet would be to set the option to "use current colour settings" when creating the rule. So you set the colour you want first, then create the rule. Here's the possible catch, remember that the transition time needs to be a value we can set, 60 min by default but adjustable - any lights that are powered on (and hence become reachable) part way through the transition time need to pick up the current transition point. So if we are 30 mins into the 60 min transition, when the unreachable lights are powered on and become reachable, they would change to the same "half way" transition to match the rest of the lights and continue from there until fully transitioned.

Does this make sense?

**Comment by Hyrules** [comment]
Yes. I understand. it's certainly doable. The tool itself would have to monitor the transition so It would compensate. Ideally the best thing would be to configure the tool for a specific set of light and have a watchdog check the light on state on a few second basis. Worst case it would take something like 15 seconds for a light to change to the specified color when powered on.

**Comment by Sandown** [comment]
Sounds good. The possible 15 second delay is fine. So are you feeling up to the challenge? :)

**Comment by Hyrules** [comment]
I will add this to the requested feature. I cannot promise I will do this soon. I want to make WinHue mostly bug free before adding any major feature.
